Refactor/mypage

chichoon opened the pull request:
## 변경 사항

- 폴더구조 개편 (components, styles, utils, hooks, etc...)
- 기타 더러운 코드들 (중첩된 삼항연산자, 불필요한 &&연산자 등...) 제거 및 깔끔한 코드 만들기
- 절대경로 Import 지향

## 변경한 이유

- 아토믹 디자인 너무 싫어요
- 한달 전의 나의 코드가 무척이나 더러움

## 스크린샷 또는 관련 문서

[리팩토링 정리](https://chichoon.tistory.com/621)

코드보다는 어떤 식으로 리팩토링하고자 했는지 개인적인 생각이 더 많습니다

hyongti reviewed (approved):
오홍 저런 식으로 컴포넌트를 통해 리다이렉트를 시킬 수 있군용 ㅎㅎ

chichoon committed:
Rename: organisms 폴더 안에 내용물 pages 폴더로 옮기기

chichoon committed:
Feat: Mypage 내에서 사용되는 버튼 스타일 _common 폴더에 분리

chichoon committed:
Refactor: 폴더 구조 개편 후 오류 나지 않게 import 절대경로로 수정

GulSam00 commented on src/components/pages/Mypage/MyProfile/ProfileSection/AuthButton.jsx:
```diff
@@ -1,8 +1,8 @@
 // import { authState } from "../../../store/auth";
-import { MypageButton } from '../../atoms/Mypage';
+import { MypageButton } from '../../_common';
 
```
이 store만 절대경로가 아니라 상대경로로 접근해서 import 했는데 이유가 있을까요?

chichoon replied:
앗 이것은... vscode의 자동 import 때문입니다
지금 하나씩 리팩토링하면서 왠만하면 절대경로로 교체하고 있어요

GulSam00 reviewed (approved):
수고하셨습니다. 훨씬 깔끔하네요

PIut0 committed:
Refactor: Writing 페이지 디렉토리 변경

PIut0 committed:
Refactor: header 에 토큰을 강제로 넣는 코드 주석처리

blingblin-g committed:
Test: first test for loginButton component in TopNav

Co-authored-by: rockpell <aythffk@gmail.com>

rockpell committed:
Chore: setupTests 추가

- test 파일에서 매번 import 하지 않아도 되는 setupTests 파일 추가
- cra로 생성된 프로젝트는 별도의 설정 파일 없이 setupTest 파일만 있으면 적용 가능

chichoon committed:
Refactor: tsconfig.json에 prettier 적용됨 (흠)

chichoon committed:
Rename: 대대적인 파일 위치 공사 시작 .......

chichoon committed:
Fix: MypageButton 폴더 이동하면서 생긴 문제 전부 해결

chichoon committed:
Feat: Mypage 메인 컴포넌트 리팩토링

chichoon committed:
Refactor: 아이콘들 각각 개별파일로 분리

chichoon committed:
Refactor: 마이페이지 링크 섹션 리팩토링, 폴더구조 변경 및 스타일 분리

chichoon commented:
# 메모
LinkBox, MyProfile의 링크 섹션, MyArticlePreview, MypageButton 리팩토링 완료
갈길이 먼 상태

chichoon committed:
Rename: 마이페이지 버튼 대소문자 임시

chichoon committed:
Rename: MypageButton 대소문자 변경

chichoon committed:
Rename: MyPage => Mypage 대소문자

blingblin-g committed:
Test: Rename LoginButton

blingblin-g committed:
Test: test ci 적용

chichoon committed:
Refactor: ProfileSection의 모든 요소들 정리

chichoon committed:
Refactor: MypageProfile의 모든 요소들 정리

chichoon committed:
Refactor: MyArticleBoard 정리, 내부 프롭 간결화, 스타일 분리

chichoon committed:
Refactor: MyArticlePreview 굳이 필요없는 && 연산자 제거

chichoon committed:
Fix: default 내보내기 빠트린거 수정

chichoon committed:
Refactor: CharSelectModal 스타일 분리

chichoon committed:
Remove: ProfileSection 필요없는 import 전부 삭제

chichoon committed:
Refactor: index.js 내보내기 개편

chichoon committed:
Fix: Mypage에서 불러오기 이름 제대로 수정

chichoon commented:
# <메모>
## 중간정산 
폴더구조 개편 및 코드 리팩토링 완료, 어째서인지 데이터가 제대로 안받아와지고 있는 중 (`[baselink]/undefined/users/me` 로 전송을 보내고 있음 ??)

## 해야할 것
훅 분리 (기나긴 로직들 전부 커스텀훅으로 분리 후 필요한것만 빼오기)
네트워크 뭐가 문젠지 보기
상대경로 -> 절대경로로 변경, 임포트 순서 맞추기

chichoon committed:
Feat: 모달 구현 (react-portal)

chichoon committed:
Refactor: CharSelectModal mui 안쓰도록 개편

chichoon committed:
Fix: useState 안 불러왔던 문제 및 useEffect 의존성 빠진 문제 해결

chichoon committed:
Refactor: 변수명 직관적으로 변경

chichoon committed:
Refactor: mui 쓰지 않도록 변경

chichoon committed:
Refactor: 캐릭터 선택 모달 변수명 수정, 스타일 약간 수정

chichoon committed:
Refactor: MypageProfile Hook으로 분리, 에러방지용 기본값 추가

chichoon committed:
Refactor: ProfileSection Hook으로 분리

chichoon committed:
Refactor: 캐릭터 선택 모달 로직부분 정리

chichoon committed:
Fix: 캐릭터 선택부분 버튼 기본스타일 적용된거 삭제

chichoon committed:
Refactor: 프로필 영역 커스텀훅 정리 (필요한 정보 추가

chichoon committed:
Refactor: 캐릭터 선택 모달 로직 전부 제거, 프로필섹션 훅으로 넘김

chichoon committed:
Rename: profilePhotoUtils

chichoon committed:
Refactor: import 절대경로

chichoon committed:
Refactor: linkbox 없애고 내용 프로필섹션에 합침

chichoon commented:
handleClick~~ 함수명을 리팩토링하다가 전부 handle~~Click으로 바꿔버렸네요 (오마이갓)
급한불 다 끄고 수정하겠습니다,, (메모)

chichoon committed:
Rename: constants.js Mypage의 최상위폴더로 이동

chichoon committed:
Refactor: 게시글 보드 로직 훅으로 분리, 페이지셀렉터 합침, 스타일 분리

chichoon committed:
Refactor: Page Selector 스타일 분리할 필요 없다 판단

chichoon committed:
Refactor: Linkbox 스타일 분리된거 합침 (굳이 분리할 필요가 있나)

chichoon committed:
Refactor: 스타일 순서변경

chichoon committed:
Rename: profle => profile 오타...

chichoon committed:
Refactor: import 순서

chichoon committed:
Refactor: MyArticlePreview 로직부분 훅으로 분리

chichoon committed:
Refactor: Mypage 로직 훅으로 분리

chichoon committed:
Rename: handle*Click 에서 handleClick*으로 변경

chichoon committed:
Remove: 안쓰는 import 삭제

chichoon commented:
# 메모
## 중간정산
- 로직은 훅으로 분리하는 작업 완료
- 임포트 순서 및 절대경로로 임포트 수정 완료
## 해야할 것
- 알파서버 구동방법을 알게 되어 이제 정상적인 통신이 가능해졌다 (영훈님이도와주심~~ 와~~~ 영훈님 짱~~ 백엔드 짱짱~~~)
- 현재 에러방지를 위해 기본값을 설정해놔서 데이터가 제대로 안 불러와지는 곳이 많음 => 수정필요
- 데이터 받아오는 부분 에러 다 없애기
- 타입스크립트? <- 일단은 보류

chichoon committed:
Refactor: 로그인 정보는 Recoil 대신 useQuery 사용

chichoon committed:
Design: 캐릭터 선택할때 마우스 올리면 밝기 밝아지게

chichoon committed:
Feat: 모달 바깥 클릭하면 모달 닫히게 하는 훅 추가

PIut0 committed:
Feat: Category 호출

카테고리를 전역상태로 관리하지않고 페이지 내부에서 호출 및 처리
close: #67

PIut0 committed:
Refactor: CreateArticle return 타입 promise 추가

PIut0 committed:
Feat: Writing 페이지 접근시 게시판 자동 선택

PIut0 committed:
Feat: 게시판 리스트 필터링

유저의 권한에 따라 선택할 수 있는 게시판 리스트 변경

chichoon committed:
Refactor: input 순서

chichoon committed:
Feat: 조건부 setstate (잘못하면 무한리렌더링 걸림)

chichoon committed:
Refactor: 모달에서 react 라이브러리 불러오지 않도록 변경

PIut0 committed:
Design: DOM 구조 변경 및 배경 이미지 추가

rockpell commented on src/components/pages/Mypage/components/MyArticlePreview.jsx:
```diff
@@ -0,0 +1,31 @@
+          title={articleType === constants.COMMENT ? article.content : article.title}
+          likeCount={articleType ? '' : article.commentCount}
+          commentCount={articleType ? '' : article.commentCount}
+        /> /* TODO: articleType 들어간 삼항연산자 (47 ~ 48번째 줄) 테스트 필요 */
```
이 주석은 어떤 의미인걸까요
47 ~ 48번째 줄이 없는 파일인거 같아서요

chichoon replied:
아 리팩토링 전에 47~48번째줄이었는데 로직을 훅으로 아예 분리하면서 줄 수가 줄어들었는데 주석은 수정을 못했네요...
likecount, commentCount 계산하는 부분 삼항연산자를 의미합니다

rockpell commented on src/components/pages/Mypage/hooks/useMyArticlePreview.js:
```diff
@@ -0,0 +1,32 @@
+    [constants.ARTICLE]: { link: 'article', fetchFunc: UserService.getMyArticles(1), title: '내 게시글' },
+    [constants.COMMENT]: { link: 'comment', fetchFunc: UserService.getMyComments(1), title: '내 댓글' },
+    [constants.LIKED]: { link: 'liked', fetchFunc: UserService.getLikeArticles(1), title: '좋아요한 글' },
+  }[articleType] || { link: '', fetchFunc: undefined, title: '' };
```
이 코드가 useMyArticleBoard에도 있어서 중복인거 같은데
다른 부분이 있는걸까요?
인줄 알았는데 link가 추가되어 있군요
이거는 공통으로 처리할수 있으면 좋을거 같네요

chichoon replied:
이부분 한번 줄이고 utils로 분리해 보겠습니다😄

rockpell commented on src/components/pages/Mypage/hooks/useMypage.js:
```diff
@@ -0,0 +1,17 @@
+      article: constants.ARTICLE,
+      comment: constants.COMMENT,
+      liked: constants.LIKED,
+    }[location.pathname.split('/')[2]] || 0;
```
이미 알고 계실거 같지만
location.pathname.split('/')[2] 이부분은 useParam으로 바꾸면 좋을거 같습니다
기존에 이렇게 되있는 코드가 전체적으로 많아서 보일때마다 바꾸면 좋겠네요

PIut0 committed:
Design: 로그인 블럭 디자인 적용

done: #57

rockpell commented on src/components/pages/Mypage/hooks/useMypageProfile.js:
```diff
@@ -0,0 +1,60 @@
+  character: 0,
+  createdAt: '1900-01-01T00:00:00.000Z',
+  updatedAt: '1900-01-01T00:00:00.000Z',
+};
```
이거는 지금은 안쓰고 있는것 같은데 원래 쓰던걸 안쓰게 된건가요?

chichoon replied:
알파서버 구동 전에 에러 방지용으로 넣은 초기값 데이터입니다
지금은 알파서버가 정상적으로 구동되고있기도 하고 배포 후에는 로그인하지 않으면 마이페이지에 접근하지 못하게 하는 방식 등을 사용하면 초기값이 필요없을 것 같긴 하네요
한번 살펴보고 날려보겠습니다

rockpell reviewed (approved):
훨씬 깔끔해져서 좋네요
리팩토링하느라 고생하셨습니다
styled 컴포넌트 내에서 className으로 사용하고 있는 부분때문에 좀헷갈리네요
나중에는 styled 컴포넌트 내부에서 class를 넣는 부분은 최대한 없애는게 좋을거 같습니다

리팩토링 정리글 잘봤습니다

PIut0 committed:
Fix: 내부 모듈 import 경로 수정

상대경로 => 절대경로

PIut0 committed:
Fix: Writing Page index 추가

chichoon committed:
Fix: MyArticlePreview 이상한 삼항연산자 제대로 동작하도록 수정

chichoon committed:
Refactor: MyArticeBoard, MyArticlePreview 공통부분 합침

chichoon committed:
Fix: 코멘트 한국어만 두번씩 달리는 현상 수정

chichoon committed:
Fix: 존재하지 않는 페이지 error로 리디렉션, 마이페이지 params로 수정

chichoon committed:
Remove: 안쓰는 초기값 삭제

blingblin-g commented on src/components/organisms/article/Comment.jsx:
```diff
@@ -43,7 +40,8 @@ const CreateComment = ({ articleId, onCreateComment }) => {
-  const handleSubmitComment = async e => {
+
+  const handleSubmitComment = async () => {
     //e.preventDefault();
```
이 주석 처리 된 부분은 지워도 되는 걸까요?

chichoon replied:
이거 다른 코드는 제가 작성한게 아니라서 혹시몰라 다른부분은 거의 손대지 않았습니다 
버튼 두번 눌리는 문제가 e때문에 그런가 싶어 e만 날려줬는데 주석은 신경쓰질 못했네요 🙊

blingblin-g commented on src/components/pages/Mypage/hooks/useClickOutside.js:
```diff
@@ -0,0 +1,19 @@
+  const handleClickOutside = () => {
+    if (ref.current && !ref.current.contains(event.target)) onClicKOutside();
+  };
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [ref]);
```
modal을 위해 `addEventListener`를 하셨군요~

chichoon replied:
이제 모달을 직접 만들수있는 사람으로 성장했습니다

rockpell commented on src/components/pages/Mypage/utils/myArticlePreviewUtils.js:
```diff
@@ -0,0 +1,14 @@
+import { UserService } from 'network';
+import constants from 'components/pages/Mypage/constants';
+
+const getArticleInfo = articleType => {
+  return (
+    {
+      [constants.ARTICLE]: { link: 'article', fetchFunc: UserService.getMyArticles(1), title: '내 게시글' },
+      [constants.COMMENT]: { link: 'comment', fetchFunc: UserService.getMyComments(1), title: '내 댓글' },
+      [constants.LIKED]: { link: 'liked', fetchFunc: UserService.getLikeArticles(1), title: '좋아요한 글' },
+    }[articleType] || { link: '', fetchFunc: undefined, title: '' }
+  );
+};
```
따로 분리하셨군요 👍

blingblin-g replied:
LGTM 👍

rockpell commented on src/components/organisms/article/Comment.jsx:
```diff
@@ -57,11 +55,14 @@ const CreateComment = ({ articleId, onCreateComment }) => {
   };
+
   const onKeyDown = e => {
+    if (e.nativeEvent.isComposing) return;
```
좋은거 알아갑니다

chichoon replied:
굿굿

blingblin-g replied:
저도 새롭게 배워갑니다~~

blingblin-g commented on src/components/pages/Mypage/styles/MypageProfile.styles.js:
```diff
@@ -0,0 +1,89 @@
+import styled from 'styled-components';
+
+const StyledMypageProfile = styled.div`
```
이 친구는 좀 긴 것 같은데 나중에 depth를 쪼개고 분리하는 것도 좋을 것 같아요~

chichoon replied:
아까 이야기 나온 대로 클래스명을 최대한 제거하고 depth를 줄이는 방식으로 고쳐봐야겠군요 생각보다 오래걸릴것같네요 흫

blingblin-g commented on src/components/pages/Mypage/styles/CharSelectModal.styles.js:
```diff
@@ -0,0 +1,84 @@
+import styled from 'styled-components';
+
+const StyledCharSelectModal = styled.div`
```
나중에 depth가 너무 깊지 않게 분리해주면 좋을 것 같아요~

blingblin-g commented on src/components/pages/Mypage/hooks/useProfileSection.js:
```diff
@@ -0,0 +1,53 @@
+
+import { profilePhotoUtils } from '../utils';
+
+const useProfileSection = userInfo => {
```
hook으로 로직을 분리하신 부분이 정말 좋네요 👍

blingblin-g reviewed (approved):
고생 많으셨습니다~ 👍
코드가 훨씬 더 깔끔하고 예뻐졌어요!! 😆 
보면서 저도 많이 배우고 갑니다~ 블로그도 넘 좋아요~~ ❤️

chichoon committed:
Fix: 댓글버튼 누르면 게시글이 아니라 댓글번호로 링크 이동하는 문제 수정
